### PR TITLE
Implement support for multiple params to access indv. poolpairs

### LINKED ISSUE #1059: `DEX` Multiple permalink params support
#### What would you like to be added:
Implement support for multiple params to access each individual Pool Pairs pages.

Permalink format for DEX individual page:
1. /dex/dBTC (canonical)
2. /dex/btc (aliases)
3. /dex/dBTC-DFI
4. /dex/5/

#### Why is this needed:

/triage accepted
/area dex ui-ux
/assign @nichellekoh

### PR DESCRIPTION
**Pull request by nichellekoh**
…page

#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #1059

#### Sample Links & Screenshots:

Link: 
<details>
<summary>Desktop Screenshot</summary>

image

</details>
<details>
<summary>Mobile Screenshot</summary>

</details>

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on multiple web browsers
- [ ] Tested responsiveness (e.g, iPhone, iPad, Desktop)
- [ ] No console errors
- [ ] Unit tests*
- [ ] Added e2e tests*

### CODE REVIEW

File: src/pages/dex/[poolpairId]/index.page.tsx [hunk 1]
```diff
@@ -67,15 +68,47 @@ export default function PoolPairPage (props: InferGetServerSidePropsType<typeof
...
+
+    let poolpairsResponse = await api.poolpairs.list(200)
+    poolpairs.push(...poolpairsResponse)
+    while (poolpairsResponse.hasNext) {
+      poolpairsResponse = await api.poolpairs.list(200, poolpairsResponse.nextToken)
+      poolpairs.push(...poolpairsResponse)
+    }
+    poolPair = poolpairs.filter(pair => {
+      return pair.tokenA.displaySymbol === tokenA && pair.tokenB.displaySymbol === tokenB
+    }).pop()!
```
Comment (joeldavidw) [inline src/pages/dex/[poolpairId]/index.page.tsx:92]: Try not to use `!`. Add undefined to the variable type instead and check for it

File: src/pages/dex/[poolpairId]/index.page.tsx [hunk 2]
```diff
@@ -67,15 +68,47 @@ export default function PoolPairPage (props: InferGetServerSidePropsType<typeof
...
 export async function getServerSideProps (context: GetServerSidePropsContext): Promise<GetServerSidePropsResult<PoolPairPageProps>> {
-  const poolpairId = context.params?.poolpairId?.toString().trim() as string
+  let poolpairId = context.params?.poolpairId?.toString().trim() as string
+  if (!isAlphanumeric(poolpairId, '-')) {
+    return { notFound: true }
+  }
+
   const api = getWhaleApiClient(context)
 
+  let poolPair: PoolPairData
```
Comment (joeldavidw) [inline src/pages/dex/[poolpairId]/index.page.tsx:78]:
```suggestion
  let poolPair: PoolPairData | undefined
  ```

File: src/pages/dex/[poolpairId]/index.page.tsx [hunk 3]
```diff
@@ -67,15 +68,47 @@ export default function PoolPairPage (props: InferGetServerSidePropsType<typeof
...
+      poolpairs.push(...poolpairsResponse)
+    }
+    poolPair = poolpairs.filter(pair => {
+      return pair.tokenA.displaySymbol === tokenA && pair.tokenB.displaySymbol === tokenB
+    }).pop()!
+
+    if (poolPair === undefined) {
+      return { notFound: true }
+    }
+    poolpairId = poolPair?.id
```
Comment (joeldavidw) [inline src/pages/dex/[poolpairId]/index.page.tsx:97]:
```suggestion
    poolpairId = poolPair.id
```

**Review by joeldavidw — CHANGES_REQUESTED** [review]

File: src/pages/dex/[poolpairId]/index.page.tsx (hunk 1 above)
Comment (joeldavidw) [reply src/pages/dex/[poolpairId]/index.page.tsx:92]: We should shift this whole if chunk into a function. It is going to get more complex when we add in the other url types.

Comment (nichellekoh) [reply src/pages/dex/[poolpairId]/index.page.tsx]: Pushed code to split logic into a separate async function

**Review by joeldavidw — APPROVED** [review]

**Pull request merged**